Local build failure after 2f5dde4e

## Summary

Bazel fails to build LLVM after the update to LLVM r368958 (commit 2f5dde4e) on macOS.

## Details

```
$ bazel version
Build label: 0.28.1
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Fri Jul 19 15:22:50 2019 (1563549770)
Build timestamp: 1563549770
Build timestamp as int: 1563549770
```

```
$ xcodebuild -version
Xcode 10.3
Build version 10G8
```

```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.14.6
BuildVersion:	18G87
```

## Repro

```
$ git checkout --detach 2f5dde4e
$ bazel test kythe/go/...
```

I had some trouble figuring out the exact target that triggers this; building individual targets seems to work OK for the ones I tried. The glob is obviously not minimal, but it will trigger the error reliably.

## Expected

All tests build and pass.

## Observed

```
ERROR: /private/var/tmp/_bazel_michael/4ad4e77a574cde1e20a887307ff11f34/external/org_llvm/BUILD.bazel:188:1: C++ compilation of rule '@org_llvm//:llvm-tblgen' failed (Exit 1) sandbox-exec failed: error executing command 
  (cd /private/var/tmp/_bazel_michael/4ad4e77a574cde1e20a887307ff11f34/sandbox/darwin-sandbox/2821/execroot/io_kythe && \
  exec env - \
    APPLE_SDK_PLATFORM=MacOSX \
    APPLE_SDK_VERSION_OVERRIDE=10.14 \
    DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer \
    PATH=/bin:/usr/bin:/usr/local/bin \
    SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk \
    TMPDIR=/var/folders/m8/8x5f9_qn2gs1zxr6c86jw1940000gn/T/ \
    XCODE_VERSION_OVERRIDE=10.3.0.10G8 \
  /usr/bin/sandbox-exec -f /private/var/tmp/_bazel_michael/4ad4e77a574cde1e20a887307ff11f34/sandbox/darwin-sandbox/2821/sandbox.sb /var/tmp/_bazel_michael/install/8772b695a4a7b498a4dd3eff54505d9c/_embedded_binaries/process-wrapper '--timeout=0' '--kill_delay=15' external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG '-std=c++11' -iquote external/org_llvm -iquote bazel-out/host/bin/external/org_llvm -iquote external/net_zlib -iquote bazel-out/host/bin/external/net_zlib -iquote external/bazel_tools -iquote bazel-out/host/bin/external/bazel_tools -isystem external/org_llvm/include -isystem bazel-out/host/bin/external/org_llvm/include -isystem external/net_zlib/zlib/include -isystem bazel-out/host/bin/external/net_zlib/zlib/include -MD -MF bazel-out/host/bin/external/org_llvm/_objs/llvm-tblgen/TableGen.d '-frandom-seed=bazel-out/host/bin/external/org_llvm/_objs/llvm-tblgen/TableGen.o' -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks '-mmacosx-version-min=10.14' -g0 -g0 -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/org_llvm/utils/TableGen/TableGen.cpp -o bazel-out/host/bin/external/org_llvm/_objs/llvm-tblgen/TableGen.o)
In file included from external/org_llvm/utils/TableGen/TableGen.cpp:19:
In file included from external/org_llvm/include/llvm/TableGen/Record.h:27:
external/org_llvm/include/llvm/Support/TrailingObjects.h:252:24: error: no member named 'is_final' in namespace 'std'
    static_assert(std::is_final<BaseTy>(), "BaseTy must be final.");
                  ~~~~~^
external/org_llvm/include/llvm/Support/TrailingObjects.h:252:33: error: 'BaseTy' does not refer to a value
    static_assert(std::is_final<BaseTy>(), "BaseTy must be final.");
                                ^
external/org_llvm/include/llvm/Support/TrailingObjects.h:227:20: note: declared here
template <typename BaseTy, typename... TrailingTys>
                   ^
external/org_llvm/include/llvm/Support/TrailingObjects.h:252:41: error: expected expression
    static_assert(std::is_final<BaseTy>(), "BaseTy must be final.");
                                        ^
3 errors generated.
```

--- a/.bazelrc
+++ b/.bazelrc
@@ -12,9 +12,8 @@ build --auto_cpu_environment_group=//buildenv:cpu
 
 # Ensure clang is used, by default, over any other C++ installation (e.g. gcc).
 # LLVM requires C++14, so ensure that is used.
-build --cxxopt=-std=c++14 --host_cxxopt=-std=c++14
 build --client_env=CC=clang
-build --client_env=BAZEL_CXXOPTS=-std=c++14
+build --cxxopt=-std=c++14 --host_cxxopt=-std=c++14 --client_env=BAZEL_CXXOPTS=-std=c++14
 
 # Ensure environment variables are static across machines; allows for cross-user caching.
 build --experimental_strict_action_env

--- a/.bazelrc
+++ b/.bazelrc
@@ -12,6 +12,7 @@ build --auto_cpu_environment_group=//buildenv:cpu
 
 # Ensure clang is used, by default, over any other C++ installation (e.g. gcc).
 # LLVM requires C++14, so ensure that is used.
+build --cxxopt=-std=c++14 --host_cxxopt=-std=c++14
 build --client_env=CC=clang
 build --client_env=BAZEL_CXXOPTS=-std=c++14
 

--- a/kythe/web/site/getting-started.md
+++ b/kythe/web/site/getting-started.md
@@ -67,6 +67,7 @@ apt-get install \
 {% endhighlight %}
 
 #### Troubleshooting bazel/clang/llvm errors
+
 You must either have `/usr/bin/clang` aliased properly, or the `CC` env var set
 for Bazel:
 
@@ -103,6 +104,14 @@ then you need to clean and rebuild your TOOLCHAIN:
 {% highlight bash %}
 bazel clean --expunge && bazel build @local_config_cc//:toolchain
 {% endhighlight %}
+
+Note also that Kythe depends on LLVM, which in turn requires support for C++14.
+In most installations, C++14 is not enabled by default, so the default Kythe
+`.bazelrc` includes the necessary flag (`-std=c++14`) to enable it.
+
+If you have user-specific Bazel settings that override the defaults, you may
+need to include these flags explicitly. If you get errors about undefined C++14
+names (such as `std::is_final`), check for this.
 
 ## Building Kythe
 

